extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3InputDataConfig.java

```diff
@@ -35,7 +36,8 @@
    */
   @JsonProperty
   @Min(1)
-  private int maxListingLength = 1024;
+  @Max(1000)
+  private int maxListingLength = 1000;
```

Comment:
Are there any tests that verify the default / min and max ?

Also how is the setter used below? it doesn't look like it checks that the passed in maxListingLength is within the provided bounds.

Reply:
Addressed. Good suggestion.